src/Orchard.Environment.Shell.Abstractions/Builders/Models/ShellBlueprint.cs

```diff
@@ -12,10 +13,23 @@ namespace Orchard.Environment.Shell.Builders.Models
     /// </summary>
     public class ShellBlueprint
     {
+        private static readonly IFeatureInfo CoreFeature
+            = new InternalFeatureInfo("Core", new InternalExtensionInfo("Core"));
+
         public ShellSettings Settings { get; set; }
         public ShellDescriptor Descriptor { get; set; }
 
-        public IEnumerable<DependencyBlueprint> Dependencies { get; set; }
+        public IDictionary<Type, DependencyBlueprint> Dependencies { get; set; }
```

Comment:
We need a default ctor that takes all three properties, and make then `get;` only.